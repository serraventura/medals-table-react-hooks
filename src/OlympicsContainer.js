import React, { useReducer } from "react";
import MedalTable from "./MedalTable";
import CountryForm from "./CountryForm";
import reducer, { initialState } from "./reducer";
import { selectAllCountries } from "./selectors";
import { addMedal, addCountry } from "./actions";
import { MainContainer } from "./styledComponents";

const OlympicsContainer = () => {
  const [state, dispatch] = useReducer(reducer, initialState);
  const addMedalHandler = (country, type) => addMedal(dispatch, country, type);

  return (
    <MainContainer>
      <CountryForm onEnter={country => addCountry(dispatch, country)} />
      <MedalTable
        list={selectAllCountries(state)}
        onAddGold={addMedalHandler}
        onAddSilver={addMedalHandler}
        onAddBronze={addMedalHandler}
      />
    </MainContainer>
  );
};

export default OlympicsContainer;
