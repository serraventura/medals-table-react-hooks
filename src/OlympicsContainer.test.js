import React from "react";
import { shallow, mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import OlympicsContainer from "./OlympicsContainer";

configure({ adapter: new Adapter() });

describe("<OlympicsContainer />", () => {
  it("should render without crashing", () => {
    const wrapper = shallow(<OlympicsContainer />);
    expect(wrapper).toBeTruthy();
  });

  it("should match snapshot", () => {
    const wrapper = mount(<OlympicsContainer />);
    expect(wrapper).toMatchSnapshot();
  });  
});
