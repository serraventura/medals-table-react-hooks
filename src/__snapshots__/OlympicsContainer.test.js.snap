// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OlympicsContainer /> should match snapshot 1`] = `ReactWrapper {}`;
